src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Debug/ConsoleDebugLaunchProviderTests.cs

```diff
@@ -360,6 +358,64 @@ public void GetExeAndArgumentsWithEmptyArgs()
             Assert.Equal(@"e:\myexe.exe", targets[0].Executable);
         }
 
+        [Fact]
+        public async Task QueryDebugTargetsAsync_WhenLibraryWithRunCommand_ReturnsRunCommand()
+        {
+            var properties = new Dictionary<string, string>() {
+                {"RunCommand", @"C:\dotnet.exe"},
+                {"TargetFrameworkIdentifier", @".NETFramework" }
+                };
+
+            _mockFS.WriteAllText(@"C:\dotnet.exe", string.Empty);
+
+            var debugger = GetDebugTargetsProvider("Library", properties);
+
+            var activeProfile = new LaunchProfile() { Name = "Name", CommandName = "Project" };
+
+            var targets = await debugger.QueryDebugTargetsAsync(0, activeProfile);
+
+            Assert.Single(targets);
+            Assert.Equal(@"C:\dotnet.exe", targets[0].Executable);
+        }
+
+        [Fact]
+        public async Task QueryDebugTargetsAsync_WhenLibraryWithoutRunCommand_ReturnsTargetPath()
+        { 
+            var properties = new Dictionary<string, string>() {
+                {"TargetPath", @"C:\library.dll"},
+                {"TargetFrameworkIdentifier", @".NETFramework" }
+                };
+
+            _mockFS.WriteAllText(@"C:\library.dll", string.Empty);
+
+            var debugger = GetDebugTargetsProvider("Library", properties);
+
+            var activeProfile = new LaunchProfile() { Name = "Name", CommandName = "Project" };
+
+            var targets = await debugger.QueryDebugTargetsAsync(0, activeProfile);
+
+            Assert.Single(targets);
+            Assert.Equal(@"C:\library.dll", targets[0].Executable);
+        }
+
+        [Fact]
+        public async Task QueryDebugTargetsForDebugLaunchAsync_WhenLibrary_Throws()
```

Comment:
So, the difference between this and the test before is just whether the library exists on disk?  Is that right, or should this return the targetpath anyway, assuming that something will cause a build in between?

Reply:
Ah, no it's not. This test is showing the previous behavior (that know only occurs during QueryDebugTargetsForDebugLaunch), but given I forgot to create the file on disk it could be either, I'll change the test.